src/react-queries/helpers.ts

```diff
@@ -0,0 +1,62 @@
+import type { UseQueryResult } from '@tanstack/react-query'
+
+/**
+ * Merges react-query outputs into a into a single output. Use with `useQueries` and `combine`.
+ *
+ * @param queryOutputs The react-query `useQueries` outputs to merge.
+ * @param combineResults The function to combine the result data of the queries.
+ * @returns The merged react-query output.
+ */
+export const mergeQueryOutputs = <QueryOutputType, QueryErrorType, MergedOutputType>(
```

Comment:
@gomesalexandre  gm 🎉

Reply:
Looks perf! 
Can we add `isPaused` in a follow-up so we can handle loading states even more gracefully here @woodenfurniture? 🙏🏽